galsim/lsst/lsst_wcs.py

```diff
@@ -0,0 +1,1164 @@
+
+        a list of y pixel coordinates on the chip specified for this WCS
+        """
+        camera_point_list = self._camera._get_afw_pupil_coord_list_from_float(ra, dec)
+        xx, yy = \
+        self._camera._pixel_coord_from_point_and_name(camera_point_list,
+                                                      [self._chip_name]*len(camera_point_list))
+
+        if len(xx)==1:
+            return xx[0], yy[0]
+        else:
+            return xx, yy
+
+
+    def _radec(self, x, y):
+        """
+        inputs
+        ------------
+        x is the x pixel coordinate on this chip (can be a list)
+
+        y is the y pixel coordinate on this chip (can be a list)
+
+        outputs
+        ------------
+        ra is in radians
+
+        dec is in radians
+        """
+
+        if hasattr(x, '__len__'):
+            chip_name = [self._chip_name]*len(x)
+        else:
+            chip_name = self._chip_name
+
+        return self._camera.raDecFromPixelCoords(x, y, chip_name)
+
+
+    def _getTanPixelBounds(self):
+        """
+        Return the minimum and maximum values of x and y in TAN_PIXELS
+        coordinates (pixel coordinates without any distoration due to
+        optics applied).
+
+        output order is: xmin, xmax, ymin, ymax
+        """
+
+        tanPixelSystem = self._detector.makeCameraSys(TAN_PIXELS)
+        xPixMin = None
+        xPixMax = None
+        yPixMin = None
+        yPixMax = None
+        cornerPointList = self._detector.getCorners(FOCAL_PLANE)
+        for cornerPoint in cornerPointList:
+            cameraPoint = self._camera._camera.transform(
+                               self._detector.makeCameraPoint(cornerPoint, FOCAL_PLANE),
+                               tanPixelSystem).getPoint()
+
+            xx = cameraPoint.getX()
+            yy = cameraPoint.getY()
+            if xPixMin is None or xx<xPixMin:
+                xPixMin = xx
+            if xPixMax is None or xx>xPixMax:
+                xPixMax = xx
+            if yPixMin is None or yy<yPixMin:
+                yPixMin = yy
+            if yPixMax is None or yy>yPixMax:
+                yPixMax = yy
+
+        return xPixMin, xPixMax, yPixMin, yPixMax
+
+
+    def getTanWcs(self):
+        """
+        Return a WCS which approximates the focal plane as perfectly flat
+        (i.e. it ignores optical distortions that the telescope may impose on the image)
+
+        The output is an instantiation of lsst.afw.image's TanWcs class
+        representing the WCS of the detector as if there were no optical
+        distortions imposed by the telescope.
+        """
+
+        xTanPixMin, xTanPixMax, \
+        yTanPixMin, yTanPixMax = self._getTanPixelBounds()
+
+
+        xPixList = []
+        yPixList = []
+        nameList = []
+
+        #dx and dy are set somewhat heuristically
+        #setting them eqal to 0.1(max-min) lead to errors
+        #on the order of 0.7 arcsec in the WCS
```

Comment:
This comment confuses me.  The next two lines use 0.5(max-min) for dx and dy, not 0.1.  So is that more accurate than using 0.1, or less?  If less (or the same), then I think we want something better, since 0.7 arcsec is a huge error!  That's wrong by of order the full width of a typical PSF.